sno/working_copy.py

```diff
@@ -674,6 +732,25 @@ def delete_features(self, dbcur, dataset, pk_iter):
 
         return feat_count
 
+    def drop_table(self, target_tree_or_commit, *datasets):
+        with self.session() as db:
+            dbcur = db.cursor()
+            for dataset in datasets:
+                if dataset.has_geometry:
+                    self._drop_spatial_index(dataset)
+                table = dataset.name
+                dbcur.execute(f"""DROP TABLE {gpkg.ident(table)};""")
+
+        with self.session() as db:
```

Comment:
any reason these with/cursor blocks are separated?

Reply:
I thought there was but apparently not. Fixed.